omod/src/test/java/org/openmrs/module/idgen/web/controller/IdentifierSourceRestControllerTest.java

```diff
@@ -0,0 +1,133 @@
+/* * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.idgen.web.controller;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.idgen.service.IdentifierSourceService;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.test.Util;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+public class IdentifierSourceRestControllerTest extends MainResourceControllerTest {
+	
+	public static final String IDENTIFIER_SOURCE_UUID = "0d47284f-9e9b-4a81-a88b-8bb42bc0a999";
+	
+	private IdentifierSourceService service;
+
+	@Before
+	public void setUp() throws Exception {
+		executeDataSet("org/openmrs/module/idgen/include/TestDataForLoadTest.xml");
+		this.service = Context.getService(IdentifierSourceService.class);
+	}
+
+	@Override
+	public String getURI() {
+		return "idgen/identifiersource";
+	}
+
+	@Override
+	public String getUuid() {
+		return IDENTIFIER_SOURCE_UUID;
+	}
+
+	@Override
+	public long getAllCount()  {
+		return service.getAllIdentifierSources(false).size();
+	}
+	
+	@Test
+	public void shouldEditAnIdentifierSource() throws Exception {
+		
+		String description = "{\"description\":\"Updated description\"}";
+		
+		handle(newPostRequest(getURI() + "/" + getUuid(), description));
+		
+		assertEquals("Updated description", service.getIdentifierSourceByUuid(getUuid()).getDescription());
```

Comment:
How sure are you that it was not "Updated description" even before the POST?

Reply:
added `assertEquals(null, service.getIdentifierSourceByUuid(getUuid()).getDescription());` before the post request.

Reply:
Are you aware of assertNull? 😊